src/components/TabBar/CustomTabBar.tsx

```diff
@@ -0,0 +1,194 @@
+  top: undefined,
+  flex: 1,
+  alignItems: "center",
+  zIndex: 1,
+  justifyContent: "flex-end",
+})``;
+
+const Touchable = styled(TouchableOpacity)`
+  flex: 1;
+  align-items: center;
+`;
+
+const darkGradients = [
+  {
+    height: GRADIENT_HEIGHT,
+    opacity: 0.8,
+    stops: [
+      <Stop offset="0%" stopOpacity={0} stopColor="#131214" />,
+      <Stop offset="100%" stopOpacity={1} stopColor="#131214" />,
+    ],
+  },
+  {
+    height: 85,
+    opacity: 0.8,
+    stops: [
+      <Stop offset="0%" stopOpacity={0} stopColor="#131214" />,
+      <Stop offset="100%" stopOpacity={1} stopColor="#131214" />,
+    ],
+  },
+];
+
+const lightGradients = [
+  {
+    height: GRADIENT_HEIGHT,
+    opacity: 1,
+    stops: [
+      <Stop key="0%" offset="0" stopOpacity={0} stopColor="#ffffff" />,
+      <Stop key="100%" offset="100%" stopOpacity={0.8} stopColor="#ffffff" />,
+    ],
+  },
+  {
+    height: 85,
+    opacity: 0.8,
+    stops: [
+      <Stop key="0%" offset="0" stopOpacity={0} stopColor="#ffffff" />,
+      <Stop offset="57%" stopOpacity={0.15} stopColor="#000000" />,
+      <Stop key="100%" offset="100%" stopOpacity={0.15} stopColor="#000000" />,
+    ],
+  },
+];
+
+export default function CustomTabBar({
+  state,
+  descriptors,
+  navigation,
+  colors,
+  insets,
+}: any): JSX.Element {
+  const bgColor = getBgColor(colors);
+  const gradients = colors.type === "light" ? lightGradients : darkGradients;
+  const { bottom: bottomInset } = insets;
+  return (
+    <Flex
+      width="100%"
+      flexDirection="row"
+      height={56}
+      bottom={bottomInset}
+      position="absolute"
+      overflow="visible"
+    >
+      <BackgroundGradient {...gradients[0]} />
+      <BackgroundGradient {...gradients[1]} />
+      <BottomFiller bottom={-bottomInset} height={bottomInset} />
+      <BackgroundFiller left={0} />
+      <BackgroundFiller right={0} />
+      <Flex
+        flexDirection="row"
+        justifyContent="center"
+        alignItems="center"
+        position="absolute"
+        left={-1}
+        right={0}
+      >
+        <TabBarShape color={bgColor} />
+      </Flex>
+      {state.routes.map((route, index) => {
+        const { options } = descriptors[route.key];
+        const Icon = options.tabBarIcon;
+
+        const isFocused = state.index === index;
+
+        const onPress = () => {
+          const event = navigation.emit({
+            type: "tabPress",
+            target: route.key,
+            canPreventDefault: true,
+          });
+
+          if (!isFocused && !event.defaultPrevented) {
+            // The `merge: true` option makes sure that the params inside the tab screen are preserved
+            navigation.navigate({ name: route.name, merge: true });
+          }
+        };
+
+        const onLongPress = () => {
+          navigation.emit({
+            type: "tabLongPress",
+            target: route.key,
+          });
+        };
+
+        if (index === 2) {
```

Comment:
this is the middle button, we want to give it a full screen container (so it can draw the drawer & the backdrop over the full screen) that has zIndex: 1 (so it draws over the next buttons rendered in the tabBar)